src/tsserver/nodeServer.ts

```diff
@@ -127,23 +135,24 @@ function parseServerMode(): LanguageServiceMode | string | undefined {
     const mode = findArgument("--serverMode");
     if (!mode) return undefined;
 
-    switch (mode.toLowerCase()) {
-        case "semantic":
-            return LanguageServiceMode.Semantic;
-        case "partialsemantic":
-            return LanguageServiceMode.PartialSemantic;
-        case "syntactic":
-            return LanguageServiceMode.Syntactic;
-        default:
-            return mode;
-    }
+    const serverMode = {

```

Comment:
This incurs an object allocation on every call, and the object gets bigger every time we add more stuff. I get that this looks "cleaner" but there are reasons to use `switch` as well.